### ISSUE TITLE
CDN nor docker repository cannot be dettached from last variant

### ISSUE THREAD

**Issue by pbabinca** [ktdreyer/errata-tool-ansible#191]
I have a CDN repository or docker repository attached to a variant. I'd like to remove it from it. When I enter empty list of variants, it translates to API call against `/api/v1/cdn_repos/<repo_id>` with data `{"cdn_repo": {"variant_names": []}}`. Even though this seems to be correct approach according to docs:
```
If any variant parameters are specified, only the specified variants will be linked to the repository. Any variants that are currently linked but not specified in the request will be detached from the repository.
```
repositores are not detached.

To detach a repository from specific variant web UI uses undocumented API call `/variants/<variant_id>/cdn_repos/detach.json` with data  `{"cdn_repo": {"id": "<repository_id>"}}` where `<repository_id>` is actual numeric ID of the repo. Until ET fixes its behavior this might be used instead.

For a reference this is tracked in CLOUDWF-5460.

**Comment by ktdreyer** [comment]
Good catch, thanks. I agree with the workaround of using `/variants/<variant_id>/cdn_repos/detach.json`, especially if rcm-docker-errata has been using this for years. We can build in some fallback logic here, so that we only hit that undocumented endpoint if we find that we did not detach everything (in other words, as long as CWFENGINE-204 is unfixed).

**Comment by ktdreyer** [comment]
There was some confusion in CWFENGINE-204. The developers could not identify a reason that users would want to  detaching all variants from a CDN repo. Today I had to do it for Ceph in CWFCONF-11004.
